pkg/platform/openstack/networking.go

```diff
@@ -183,12 +225,53 @@ func findOpenStackRouter(client *gophercloud.ServiceClient, name, tag string) (r
 	if len(routerList) == 1 {
 		return routerList[0], nil
 	} else if len(routerList) == 0 {
-		return empty, errors.New("router not found")
+		networkPorts, err := getOpenStackPortsByNetwork(client, networkID)
+		if err != nil {
+			return empty, errors.Wrap(err, "failed to list Ports on Network")
+		}
+		for _, port := range networkPorts {
+			if port.DeviceID != "" {
+				page, err := routers.List(client, routers.ListOpts{ID: port.DeviceID}).AllPages()
+				if err != nil {
+					return empty, errors.Wrap(err, "failed to get router list")
+				}
+				routerList, err := routers.ExtractRouters(page)
+				if err != nil {
+					return empty, errors.Wrap(err, "failed to extract routers list")
+				}
+				if len(routerList) == 1 {
+					return routerList[0], nil
+				}
```

Comment:
If we have the ID of the router we can just do `routers.Get` and simplify this.

Reply:
Indeed. Thank you!